packages/falter-berlin-service-registrar/doc/falter-berlin-service-registrar.rst

```diff
@@ -0,0 +1,72 @@
+
+Möchte man den Service-Registrar nutzen, muss man ihn als Zusatzpaket nachinstallieren. Das geht im Webinterface über *System* -> *Paketverwaltung*, wo man die Pakete
+
+..
+    falter-berlin-service-registrar
+    luci-app-falter-service-registrar
+    luci-i18n-falter-service-registrar-de
+
+installiert.
+
+Falls man den Service-Registrar ausschließlich über die Kommandozeile nutzen möchte, geht auch ein ``opkg install falter-berlin-service-registrar``.
+
+Beide Varianten haben den gleicen Funktionsumfang und unterscheiden sich ausschließlich in der Bedienung.
+
+
+LuCI-App (Webinterface)
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Der Service-Registrar befindet sich in der Admin-Oberfläche unter *Freifunk* -> *Freifunkdienste registrieren*. In der App können entweder Webseiten oder Allgemeine Dienste angelegt werden.
+
+Webeiten sind Dienste, bei denen der Freifunk-Router selbst eine (statische) Webseite ausliefert. Es wird also nicht nur entsprechend OLSR konfiguriert, sondern auch der Webserver auf dem Freifunkrouter so eingestellt, dass er die Webseite ausliefert.
+
+Allgemeine Dienste sind etwas anders. Sie eignen für Dienste, die auf einem anderen Gerät im lokalen Freifunknetzwerk laufen. Wenn man z.B. eine Webcam bereitstellen möchte, oder irgendeine andere Art von Server hat, eignet sich diese Art von Diensten am besten dafür.
+
+Am besten sollte die IP-Adresse des entsprechenden Gerätes dann auf statisch gestellt werden (unter *Status* -> *Übersicht* und dann im Abschnitt *Aktive DHCP-Leases* auf den Knopf *Auf statisch setzen* drücken).
+
+Die benötigten Felder sind sind eigentlich relativ selbsterklärend. Nachdem die Änderungen mit dem Knopf *Speichern & Anwenden* abgespeichert wurden, beginnt der Freifunkrouter mit der Einrichtung.
+
+Bis der Service im kompletten Freifunknetz verfügbar ist, können bis zu 5 Minuten vergehen.
+
+
+Kommandozeile
+^^^^^^^^^^^^^
+
+Um einen Dienst über die Kommondozeile einzutragen, muss zuerst die Konfigurationsdatei unter ``/etc/config/ffservices`` angepasst werden und anschließend das Skript ``register-services`` ausgeführt werden.
+
+Je nachdem, ob man eine Webseite oder einen Allgemeinen Dienst eintragen möchte, gibt es zwei unterscheidliche Typen von UCI-Sections. Einmal den Typen ``website`` und einmal den Typen ``service``. Beide Typen unterscheiden sich geringfügig in ihren Feldern:
+
+..
+    config website "testsite"
+            option fqdn         "testsite.olsr"
+            option description  "Webseite auf dem lokalen Router"
+            option protocol     "tcp"
+            option port         "81"
+            option web_root     "/tmp/www/testsite"
+            option disabled     0
+
+..
+    config service "extern"
+            option fqdn         "extern.olsr"
+            option description  "Ein Dienst der auf einem anderen Geraet lauft"
```

Comment:
darf ich hier auch kein ä verwenden?

Reply:
OLSR kann Service-Beschreibungen mit ä im Inhalt nicht verarbeiten.

```sh
ffserviced: registering service with
    FQDN.................: testservice.olsr
    Name/Description.....: Ein Dienst mit ä und test
    Protocol.............: tcp
    Port.................: 80
    IP Address...........: 10.31.142.73
olsrd: /etc/init.d/olsrd: warning_invalid_value() in option 'olsrd.cfg034e54.service', skipped
olsrd: /etc/init.d/olsrd: olsrd_setup_smartgw_rules() Notice: Inserting firewall rules for SmartGateway
```

Im Webinterface werden Sonderzeichen für das Eintragen abgelehnt. Am besten vermerke ich das nochmal in der Doku gesondert, was genau erlaubt ist.

![grafik](https://user-images.githubusercontent.com/43141307/216624035-84ecd645-9f32-46b8-a15c-d3e5f321e381.png)